src/main/java/io/confluent/idesidecar/restapi/models/ConnectionStatus.java

```diff
@@ -6,21 +6,145 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import io.confluent.idesidecar.restapi.auth.AuthErrors;
 import io.confluent.idesidecar.restapi.models.ConnectionStatus.Authentication.Status;
+import io.soabase.recordbuilder.core.RecordBuilder;
+import jakarta.validation.constraints.Null;
 import java.time.Instant;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
 
 /**
  * Represents the status of a connection.
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
-public record ConnectionStatus(@JsonProperty(required = true) Authentication authentication) {
+@JsonInclude(Include.NON_NULL)
+@RecordBuilder
+public record ConnectionStatus(
+    @JsonProperty("ccloud")
+    CCloudStatus ccloud,
+
+    @JsonProperty("kafka_cluster")
+    KafkaClusterStatus kafkaCluster,
+
+    @JsonProperty("schema_registry")
+    SchemaRegistryStatus schemaRegistry
+) implements ConnectionStatusBuilder.With {
+
+  // TODO: Remove this once the extension has been updated to use the new status
+  @JsonProperty(required = true)
+  public Authentication authentication() {
+    if (ccloud != null) {
+    return new Authentication(
```

Comment:
Needs indentation fix

Reply:
Good catch. I'm surprised checkstyle didn't get this. We probably need to look into that; logged as #163